lib/sles4sap/ipaddr2.pm

```diff
@@ -60,11 +72,28 @@ our $nat_pub_ip = DEPLOY_PREFIX . '-nat_pub_ip';
 # Storage account name must be between 3 and 24 characters in length
 # and use numbers and lower-case letters only.
 our $storage_account = DEPLOY_PREFIX . 'storageaccount';
-our $priv_ip_range = '192.168.';
-our $frontend_ip = $priv_ip_range . '0.50';
+our %priv_net_address_range = get_private_ip_range();
+our $priv_ip_range = ($priv_net_address_range{vnet_address_range} =~ /^(\d+\.\d+\.\d+)\./) ? $1 : '';
```

Comment:
is it possible to avoid regexp here and at line 1060? If we really need them we can move them to a function (private internal to this lib, not exported) to have them in a single place.

Reply:
I modified it. Please have a look if new code make sense.